client/src/Components/EditPostPopup.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import Modal from 'react-bootstrap/Modal'
+import Button from 'react-bootstrap/Button'
+import { useState } from 'react';
+import Input from './Input';
+import Textarea from './Textarea';
+
+export default function EditPostPopup({show , onClose, post, onUpdate}) {
+    
+  const [editedPost, setEditedPost] = useState(post)
+  
+  function handleChange (event) {
+    const { value, name } = event.target;
+    setEditedPost((prevValue) => ({ ...prevValue, [name]: value }));
+  }
+
+  const editThePost= async (post) => {
+    try {
+      const body = { ...post };
+
+      const response = await fetch(`http://localhost:1212/${post.id}`, {
+        method: "PUT",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(body)
+      })
+
+      if (response.ok) {
+        onUpdate(post);
+      } else {
+        console.error('Failed to update the contact');
+      }
+
+    } catch (error) {
+      console.error(error.message)
+    }
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    editThePost(editedPost);
+  }
+
+  function formatDate(dateString) {
+    const date = new Date(dateString);
+    const year = date.getUTCFullYear();
+    // getMonth() returns month from 0-11, so add 1 to get the correct month
+    const month = String(date.getUTCMonth() + 1).padStart(2, '0');
+    const day = String(date.getUTCDate()).padStart(2, '0');
+  
+    return `${year}-${month}-${day}`;
+  }
+  
+  function enter(){
+    console.log( "Modal is Shown" ) 
+    post.updated_at = formatDate(post.created_at)
+    setEditedPost(post)
```

Comment:
```suggestion
    const update = { ...post };
    update.updated_at = formatDate(post.created_at)
    setEditedPost(update)
```

Whenever you're making edits to objects that come in either from props or state, _that's_ when you need to have a shallow copy of the object made.

Reply:
I will change it now but I would like to talk more about it in our meeting :)